test/framework/src/main/java/org/elasticsearch/node/MockNode.java

```diff
@@ -78,6 +80,18 @@
 public class MockNode extends Node {
 
     private static class MockServiceProvider extends NodeServiceProvider {
+        @Override
+        TaskManager newTaskManager(
+            PluginsService pluginsService,
+            Settings settings,
+            ThreadPool threadPool,
+            Set<String> taskHeaders,
+            Tracer tracer,
+            String nodeId
+        ) {
+            return MockTaskManager.create(settings, threadPool, taskHeaders, tracer, nodeId);
```

Comment:
I think we should wrap this in a `if (pluginsService.filterPlugins(MockTransportService.TestPlugin.class).findAny().isEmpty())` check similar to how `newTransportService` does it.